go/service/main.go

```diff
@@ -102,16 +108,21 @@ func (d *Service) Handle(c net.Conn) {
 		logReg = newLogRegister(d.logForwarder, d.G().Log)
 		defer logReg.UnregisterLogger()
 	}
-
-	if err := d.RegisterProtocols(server, xp, connID, logReg, d.G()); err != nil {
+	shutdowners, err := d.RegisterProtocols(server, xp, connID, logReg, d.G())
+	defer func() {
```

Comment:
Does this also work when the service is shutdown for non-CLI connections?  i.e., do we stop server.Run() cleanly?